README.md

```diff
@@ -71,17 +108,31 @@ We can now directly execute our `Hello World` application on a real connected de
 $ cargo apk run
 ```
 
+If the crate includes a runnable binary as suggested above, you will likely be greeted by the following error:
+
+```console
+$ cargo apk run
+error: extra arguments to `rustc` can only be passed to one target, consider filtering
+the package by passing, e.g., `--lib` or `--bin NAME` to specify a single target
+Error: Command `cargo rustc --target aarch64-linux-android -- -L hello_world_android/target/cargo-apk-temp-extra-link-libraries` had a non-zero exit code.
+```
+
+To solve this, add `--lib` to the run invocation, like so:
+```console
+$ cargo apk run --lib
+```
```

Comment:
I haven't investigated this properly yet, but perhaps we can force-add `--lib` to our `rustc` invocation just in case?

Additionally, running `cargo apk run --bin(s, or the name of a binary target)` here results in a successful cargo build but finally fails packaging:

```
Error: Path `"hello_world_android/target/aarch64-linux-android/debug/libhello_world_android.so"` doesn't exist.
```

I think we should assert/error in `cargo-apk` when the artifacts from `cargo-subcommand` include non-library targets, but I'm not sure if we can distinguish them at this point.

Reply:
Not going to worry about this for now, behaviour will probably be revamped/improved in `xbuild`.